sql2.txt

```diff
@@ -18,29 +35,73 @@ SUM(Quantity) AS TotalQuantity and you must GROUP BY CustomerID, ProductID.  Pas
 SQL statement below.
 
 
+SELECT c.CustomerName, p.ProductName,SUM(od.Quantity) AS TotalQuantity
+FROM Customers c
+INNER JOIN Orders o ON c.CustomerID = o.CustomerID
+INNER JOIN OrderDetails od ON o.OrderID = od.OrderID
+INNER JOIN Products p ON od.ProductID = p.ProductID
+GROUP BY c.CustomerID, p.ProductID, c.CustomerName, p.ProductName
+ORDER BY c.CustomerName;
+
+###################################
+
 
+###################################
 4. Extend the previous query to give the total price as well as the customer name, product name, 
 and total quantity.  Here you will have
 (SUM(Quantity) * Price) AS TotalPrice.  Paste your SQL statement below.
 
+SELECT c.CustomerName, p.ProductName,SUM(od.Quantity) AS TotalQuantity, (SUM(od.Quantity) * p.Price) AS TotalPrice
+FROM Customers c
+INNER JOIN Orders o ON c.CustomerID = o.CustomerID
+INNER JOIN OrderDetails od ON o.OrderID = od.OrderID
+INNER JOIN Products p ON od.ProductID = p.ProductID
+GROUP BY c.CustomerID, p.ProductID, c.CustomerName, p.ProductName, p.Price
+ORDER BY c.CustomerName;
 
+###################################
+
+
+###################################
 5. Add a new Order to the Orders table.  You will have to specify the ID of an existing customer
 to satisfy the foreign key constraint.  You will also have to specify the ID of an existing
 employee.  Paste your SQL statement below.
 
 
+INSERT INTO Orders (OrderID, CustomerID, EmployeeID, OrderDate, ShipperID)
+VALUES (10444, 1, 1, "2/12/1997 ", 1)
```

Comment:
We do not need to manually assign the PRIMARY KEY when INSERTing records. SQLite automatically assigns the primary key for us, so we should leave it to our db to handle this:

> _On an [`INSERT`](https://www.sqlite.org/lang_insert.html), if the `ROWID` or `INTEGER PRIMARY KEY` column is not explicitly given a value, then it will be filled automatically with an unused integer, usually one more than the largest `ROWID` currently in use._ [See: [SQLite Docs - _Autoincrement_](https://www.sqlite.org/autoinc.html)]

```suggestion
INSERT INTO Orders (CustomerID, EmployeeID, OrderDate, ShipperID)
VALUES (1, 1, "2/12/1997 ", 1)
```